apps/api/src/app/rate-limiting/usecases/get-api-rate-limit-service-maximum-config/get-api-rate-limit-service-maximum-config.spec.ts

```diff
@@ -0,0 +1,43 @@
+import { Test } from '@nestjs/testing';
+import { GetApiRateLimitServiceMaximumConfig } from './get-api-rate-limit-service-maximum-config.usecase';
+import {
+  ApiRateLimitCategoryEnum,
+  ApiRateLimitServiceMaximumEnvVarFormat,
+  ApiServiceLevelEnum,
+  DEFAULT_API_RATE_LIMIT_SERVICE_MAXIMUM_CONFIG,
+} from '@novu/shared';
+import { expect } from 'chai';
+
+describe('GetApiRateLimitServiceMaximumConfig', () => {
+  let useCase: GetApiRateLimitServiceMaximumConfig;
+
+  beforeEach(async () => {
+    const moduleRef = await Test.createTestingModule({
+      providers: [GetApiRateLimitServiceMaximumConfig],
+    }).compile();
+
+    useCase = moduleRef.get<GetApiRateLimitServiceMaximumConfig>(GetApiRateLimitServiceMaximumConfig);
+  });
+
+  it('should use the default API rate limits when no environment variables are set', () => {
+    expect(useCase.default).to.deep.equal(DEFAULT_API_RATE_LIMIT_SERVICE_MAXIMUM_CONFIG);
+  });
+
+  it('should override default API rate limits with environment variables', () => {
+    const mockOverrideRateLimit = 65;
+    const mockRateLimitServiceLevel = ApiServiceLevelEnum.FREE;
+    const mockRateLimitCategory = ApiRateLimitCategoryEnum.GLOBAL;
+
+    const envVarName: ApiRateLimitServiceMaximumEnvVarFormat = `API_RATE_LIMIT_MAXIMUM_${
+      ApiServiceLevelEnum.FREE.toUpperCase() as Uppercase<ApiServiceLevelEnum>
+    }_${ApiRateLimitCategoryEnum.GLOBAL.toUpperCase() as Uppercase<ApiRateLimitCategoryEnum>}`;
+    process.env[envVarName] = mockOverrideRateLimit.toString();
+
+    // Re-initialize the defaults after setting the environment variable
+    useCase.loadDefault();
+    const result = useCase.default;
+
+    expect(result[mockRateLimitServiceLevel][mockRateLimitCategory]).to.equal(mockOverrideRateLimit);
+    delete process.env[envVarName]; // cleanup
+  });
+});
```

Comment:
This was a refactor from the following file (below in PR) that wasn't picked up correctly by git:
https://github.com/novuhq/novu/pull/4867/files#diff-7e5c0068062244f3c27665cfd6f282caa1cdef7ab4eb223432c264999e091984.

It is renamed `get-default-api-rate-limits` -> `get-api-rate-limit-service-maximum-config` to better reflect the fact that a configuration is being read from the environment variables.